src/core/Akka.Cluster.Tests.MultiNode/LeaderElectionSpec.cs

```diff
@@ -0,0 +1,248 @@
+                _config.Third,
+                _config.Forth)
+                .Sort(new RoleNameComparer(this));
+        }
+
+        public void ShutdownLeaderAndVerifyNewLeader(int alreadyShutdown)
+        {
+            var currentRoles = _sortedRoles.RemoveAt(alreadyShutdown);
+            currentRoles.Count.ShouldBeGreaterOrEqual(2);
+            var leader = currentRoles.First();
+            var aUser = currentRoles.Last();
+            var remainingRoles = currentRoles.Skip(1).ToImmutableList();
+            var n = "-" + (alreadyShutdown + 1);
+
+            if (Myself == _config.Controller)
+            {
+                EnterBarrier("before-shutdown" + n);
+                TestConductor.Exit(leader, 0).Wait();
+                EnterBarrier("after-shutdown" + n, "after-unavailable" + n, "after-down" + n, "completed" + n);
+            }
+            else if (Myself == leader)
+            {
+                EnterBarrier("before-shutdown" + n, "after-shutdown" + n);
+            }
+            else if (Myself == aUser)
+            {
+                var leaderAddress = GetAddress(leader);
+                EnterBarrier("before-shutdown" + n, "after-shutdown" + n);
+                MarkNodeAsUnavailable(leaderAddress);
+                AwaitAssert(() => ClusterView.UnreachableMembers.Select(x => x.Address).Contains(leaderAddress).ShouldBeTrue());
+                EnterBarrier("after-unavailable" + n);
+                Cluster.Down(leaderAddress);
+                AwaitAssert((() => ClusterView.UnreachableMembers.Select(x => x.Address).Contains(leaderAddress).ShouldBeFalse()));
+                EnterBarrier("after-down" + n, "completed" + n);
+            }
+            else if (remainingRoles.Contains(Myself))
+            {
+                var leaderAddress = GetAddress(leader);
+                EnterBarrier("before-shutdown" + n, "after-shutdown" + n);
+                AwaitAssert(() => ClusterView.UnreachableMembers.Select(x => x.Address).Contains(leaderAddress).ShouldBeTrue());
+                EnterBarrier("after-unavailable" + n);
+                EnterBarrier("after-down" + n);
+                AwaitMembersUp(currentRoles.Count - 1);
+                var nextExpectedLeader = remainingRoles.First();
+                (ClusterView.IsLeader && Myself == nextExpectedLeader).ShouldBeTrue();
+                AssertLeaderIn(remainingRoles);
+                EnterBarrier("completed" + n);
+            }
+        }
+
+        [MultiNodeFact]
+        public void AClusterOfFourNodesMustBeAbleToElectASingleLeader()
+        {
+            AwaitClusterUp(_config.First, _config.Second, _config.Third, _config.Forth);
+
+            if (Myself != _config.Controller)
+            {
+                (ClusterView.IsLeader && Myself == _sortedRoles.First()).ShouldBeTrue();
+                AssertLeaderIn(_sortedRoles);
+            }
+
+            EnterBarrier("after-1");
+            ShutdownLeaderAndVerifyNewLeader(1);
```

Comment:
The original scala code is:
```scala
    "be able to 'elect' a single leader" taggedAs LongRunningTest in {
      awaitClusterUp(first, second, third, fourth)

      if (myself != controller) {
        clusterView.isLeader should ===(myself == sortedRoles.head)
        assertLeaderIn(sortedRoles)
      }

      enterBarrier("after-1")
    }
```

Your c# has two additional calls that shouldn't be there, `ShutdownLeaderAndVerifyNewLeader(1);`  `EnterBarrier("after-3");`